src/algebra/roption.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2019 Joe Cool. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Hoang Le Truong.
+
+If α is a comm_monoid  then roption α is a comm_monoid. 
+
+-/
+
+import   algebra.module data.pfun
+
+
+universes u v 
+variables {α : Type u} {β : Type v} 
+local attribute [instance] classical.prop_decidable
+
+
+namespace roption
+noncomputable theory
+
+instance [has_add α]: has_add (roption α) := ⟨ λ x y, ⟨x.dom ∧ y.dom, λ h, x.get (h.1)+ y.get (h.2)  ⟩  ⟩
+def add_def [has_add α] (x y: roption α) : x+y =⟨x.dom ∧ y.dom, λ h, x.get (h.1)+ y.get (h.2)  ⟩ :=rfl
+
+instance  [has_zero α ] :  has_zero (roption (α ))  :=⟨some (0:α)⟩
```

Comment:
Formatting: no space before the `]`.

Reply:
thank you for your information about formatting. Because it is first time me pull request, I will try to learn it.